### PR TITLE
Setup cisco spark automatic messaging

### PR DESCRIPTION
**Pull request by miroslavKovacPantheon**
For several situation we need notification about them.
For couple of them it was set to send email but that
would not be sufficient if we want to send less important
messages. Here we introduces messaging through
messageFactory class which sends email and/or spark
messages to yang catalog admin room.

Signed-off-by: miroslav.kovac <miroslav.kovac@pantheon.tech>

### CODE REVIEW

File: tools/utility/messageFactory.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        self.__post_to_spark(message)
+
+    def send_removed_temp_diff_files(self):
+        # TODO send spark message about removed searched diff files
+        pass
+
+    def send_removed_yang_files(self, removed_yang_files):
+        LOGGER.info('Sending notification about removed yang modules')
+        message = ('The following files has been removed from yangcatalog.org'
+                   ' using api: \'\'\'{}\'\'\''.format(removed_yang_files))
```
Comment (einarnn) [inline tools/utility/messageFactory.py]: You use the pattern \’\’\’ in a number of places. Not sure why. You could also do something like:

```
print(“‘’’{}’’’”.format(blah))
```

...which looks neater.

Or did you really mean to use backticks for markdown formatting?

Comment (miroslavKovacPantheon) [reply tools/utility/messageFactory.py]: I did mean to use a backticks. it should be formatted code.

Comment (einarnn) [reply tools/utility/messageFactory.py]: Ok, let me know when you’ve updated to use backticks rather than escaped single quotes.

Comment (miroslavKovacPantheon) [reply tools/utility/messageFactory.py]: Done

File: tools/utility/messageFactory.py [hunk 2]
```diff
@@ -83,26 +83,26 @@ def send_removed_temp_diff_files(self):
...
         modified_files = '\n'.join(modified_files)
-        message = ('There were new or modified platform metadata json files '
-                   'added to yangModels-yang repository, that are currently'
-                   'being processed in following paths:\n\n'
-                   '\'\'\' New json files: \n {} \n\n Modified json files:\n{}'
-                   '\'\'\''
+        message = ("There were new or modified platform metadata json files "
+                   "added to yangModels-yang repository, that are currently"
+                   "being processed in following paths:\n\n"
+                   "\'\'\' New json files: \n {} \n\n Modified json files:\n{}"
```
Comment (einarnn) [inline tools/utility/messageFactory.py]: Did you miss these ones?

File: tools/utility/messageFactory.py [hunk 3]
```diff
@@ -83,26 +83,26 @@ def send_removed_temp_diff_files(self):
...
+                   "\'\'\'"
```
Comment (einarnn) [inline tools/utility/messageFactory.py]: And these ones?

**Pull request merged**